model/infinispan/src/main/java/org/keycloak/models/sessions/infinispan/InfinispanUserSessionProvider.java

```diff
@@ -370,13 +370,13 @@ protected Stream<UserSessionModel> getUserSessionsStream(RealmModel realm, UserS
                 String[] idpAliasSessionId = predicate.getBrokerUserId().split("\\.");
 
                 Map<String, String> attributes = new HashMap<>();
-                attributes.put(UserModel.IDP_ALIAS, idpAliasSessionId[0]);
```

Comment:
Not searching by the `IDP_ALIAS` will lead to wrong results as two IDPs can have the same `IDP_USER_ID`. Below, only the first user is retrieved. 

Either keep the IDP, or iterate over all users returned.

Reply:
Not sure if needed because the predicate is filtering the sessions. I would say that is very unlikely that multiple users are going to be returned here but if so, their sessions will be filtered.

Reply:
The code might behave different from what you think: The attributes you combine here are used to pick the user, they are not used to filter the sessions. See this line of code: 

```
 UserModel userModel = userProvider.searchForUserStream(realm, attributes, 0, null).findFirst().orElse(null);
```

It uses the attributes to find users, and then picks the first from the list. This might then be the wrong one.

Reply:
Sorry, I see what you mean. Fixed.